spring-cloud-aws-parameter-store-config/src/main/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertySources.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.aws.paramstore;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagement;
+import org.apache.commons.logging.Log;
+
+import org.springframework.util.StringUtils;
+
+public class AwsParamStorePropertySources {
+
+	private final AwsParamStoreProperties properties;
+
+	private final Log log;
+
+	public AwsParamStorePropertySources(AwsParamStoreProperties properties, Log log) {
+		this.properties = properties;
+		this.log = log;
+	}
+
+	public List<String> getAutomaticContexts(List<String> profiles) {
+		List<String> contexts = new ArrayList<>();
+		String prefix = this.properties.getPrefix();
+		String defaultContext = getContext(prefix, this.properties.getDefaultContext());
+
+		String appName = this.properties.getName();
+
+		String appContext = prefix + "/" + appName;
+		addProfiles(contexts, appContext, profiles);
+		contexts.add(appContext + "/");
+
+		addProfiles(contexts, defaultContext, profiles);
+		contexts.add(defaultContext + "/");
+		return contexts;
+	}
+
+	protected String getContext(String prefix, String context) {
+		if (StringUtils.isEmpty(prefix)) {
+			return context;
+		}
+		else {
+			return prefix + "/" + context;
+		}
+	}
+
+	private void addProfiles(List<String> contexts, String baseContext, List<String> profiles) {
+		for (String profile : profiles) {
+			contexts.add(baseContext + this.properties.getProfileSeparator() + profile + "/");
+		}
+	}
+
+	public AwsParamStorePropertySource createPropertySource(String context, boolean optional,
```

Comment:
Similar to how i suggested in SM integration, we should add logging information: https://github.com/spring-cloud/spring-cloud-aws/commit/21011e629c1e176894c707368d9dc9c871743e02#diff-ab9058ccbead5f4685b9fc609e6e10808f622475492ea1b099af6994edfa3e70R80